swim/heal_partition.go

```diff
@@ -72,19 +72,35 @@ func nodesThatNeedToReincarnate(MA, MB []Change) (changesForA, changesForB []Cha
 		// If a node would be overwritten and would stop being pingable, it is
 		// not suited for merging.
 		if b.isPingable() && a.overrides(b) && !a.isPingable() {
-			changesForB = append(changesForB, Change{
-				Address:     a.Address,
-				Incarnation: a.Incarnation,
-				Status:      Suspect,
-			})
+			// take a look at the state of the member
+			member := Member{}
+			member.populateFromChange(&a)
+
+			// create the member into a change
+			change := Change{}
+			change.populateSubject(&member)
```

Comment:
You can't copy the change here because it is essential for the working of partition healing that the source information of the change is not set on a change sent to the other partition. If you add the source information on there you have the chance of a reverse full sync that will merge the partitions before they are in a merge-able state. The same goes for populating a change from a change, it is debatable if it should copy the source information of a change.

We can do two things:

1. document better why we go change->member->change
2. have a `change.scrubSourceInformation()` to explicitly remove the source information from the change here.